File: apps/ios/Wewed/Models/ShellFixtureState.swift
import SwiftUI

public enum ShellFixtureState: String, CaseIterable, Identifiable, Sendable {
    case populated = "Populated"
    case empty = "Empty"
    case loading = "Loading"
    case error = "Error"
    case offline = "Offline"
    case attention = "Attention"
    case readOnly = "Read-Only"
    case editable = "Editable"

    public var id: String { rawValue }
}

public protocol ShellStateAware {
    var fixtureState: ShellFixtureState { get }
}

File: apps/android/app/src/main/java/pro/wewed/app/models/ShellFixtureState.kt
package pro.wewed.app.models

enum class ShellFixtureState(val title: String) {
    POPULATED("Populated"),
    EMPTY("Empty"),
    LOADING("Loading"),
    ERROR("Error"),
    OFFLINE("Offline"),
    ATTENTION("Attention"),
    READ_ONLY("Read-Only"),
    EDITABLE("Editable")
}
